test/functional/feature_signet.py

```diff
@@ -22,21 +24,37 @@
     '00000020a868e8514be5e46dabd6a122132f423f36a43b716a40c394e2a8d063e1010000f4c6c717e99d800c699c25a2006a75a0c5c09f432a936f385e6fce139cdbd1a5e9964d5fae77031e7d026e0001010000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff025a51feffffff0200f2052a01000000160014aaa671c82b138e3b8f510cd801e5f2bd0aa305940000000000000000776a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf94c4fecc7daa24900473044022042309f4c3c7a1a2ac8c24f890f962df1c0086cec10be0868087cfc427520cb2702201dafee8911c269b7e786e242045bb57cef3f5b0f177010c6159abae42f646cc501000120000000000000000000000000000000000000000000000000000000000000000000000000',
 ]
 
+class SignetParams:
+    def __init__(self, challenge=None):
+        if challenge is None:
+            self.challenge = SIGNET_DEFAULT_CHALLENGE
+            self.shared_args = []
+        else:
+            self.challenge = challenge
+            self.shared_args = [f"-signetchallenge={challenge}"]
+
+def check_getblockchaininfo(node, signet_params):
+    blockchain_info = node.getblockchaininfo()
+    assert_equal(blockchain_info['chain'], 'signet')
+    assert_equal(blockchain_info['signet_challenge'], signet_params.challenge)
 
 class SignetBasicTest(BitcoinTestFramework):
     def set_test_params(self):
         self.chain = "signet"
         self.num_nodes = 6
         self.setup_clean_chain = True
-        shared_args1 = ["-signetchallenge=51"]  # OP_TRUE
-        shared_args2 = []  # default challenge
-        # we use the exact same challenge except we do it as a 2-of-2, which means it should fail
-        shared_args3 = ["-signetchallenge=522103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae"]
+
+        self.signets = [
+            SignetParams(challenge='51'), # OP_TRUE
+            SignetParams(), # default challenge
+            # default challenge as a 2-of-2, which means it should fail
+            SignetParams(challenge='522103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae')
+        ]
 
         self.extra_args = [
-            shared_args1, shared_args1,
-            shared_args2, shared_args2,
-            shared_args3, shared_args3,
+            self.signets[0].shared_args, self.signets[0].shared_args,
```

Comment:
I don't understand why this was duplicated in the original version. @kallewoof do you remember?

Reply:
Makes no sense to me. I read through the entire thread in https://github.com/bitcoin/bitcoin/pull/18267 where this was added, and it looks like I put it there right at the end before it was merged, so it may have slipped under the radar.

Anyway, my suggestion: remove the duplication. Try the tests. If the tests fail, we know this at least has a purpose.

Reply:
Oh I get it now. There's two nodes for each signet, six in total. So this is fine.

Reply:
3 sets of pairs of nodes where each pair shares their set of args? OK, that sounds reasonable.